### ISSUE TITLE
Certain dialog windows are automatically closed by Amethyst

### ISSUE THREAD

**Issue by eplanet** [ianyh/Amethyst#651]
## System
 - **Amethyst version**: 0.11.4
 - **macOS version**: 10.12.6 (16G29)

## What's the problem?
Amethyst seems to be killing a certain type of dialog window. I hope this hasn't been reported / addressed yet, but I couldn't find any related issue.

## How can it be reproduced?
Easily open an existing project structure, then try to add dependencies to it (refer to the screenshot or [official help page](https://www.jetbrains.com/help/idea/working-with-module-dependencies.html#add_dependency)).

## What applications are involved?
* IntelliJ

## Has anything fixed it, even temporarily?
Quitting Amethyst fixes it, that's why I'm reporting the issue here.

## dotfile (if you have one)
None.

## Screenshots
![View of the menu](https://ibb.co/hM4Q1G)

**Comment by kmajor** [comment]
+1 same problem.  It's essentially killing all my chrome dialogs, which means I have to choose between Amethyst and Chrome... :-(

I've tried with both the latest chromium 63.xxxx  and latest Chrome I think 56.xxxx

To reproduce, simply click on any chrome extensions that have a dialog popup and when you mouseover, the window disappears.  

I hope you guys can get this fixed right away, it's probably not widely reported because people can't figure out what's causing it.  I spent hours tracking this down.  

Amethyst is pretty great btw...

**Comment by ianyh** [comment]
This sounds like a problem with focus-follows-mouse. Do you have that option enabled?

**Comment by eplanet** [comment]
That could have explained it but I don't have that option enabled. I tried activating it without success, the same being true for `Mouse follows focused windows` and actually any other option.

**Comment by spenserw** [comment]
I can confirm the problem is with `focus-follows-mouse`. Disabling that option no longer causes the dialogues to close. Same versions as OP.

**Comment by deiga** [comment]
Turning off `focus-follows-mouse` did not work for me. `mouse-follows-focus`could be the culprit for me, but I really need that to stop me from getting Tenosynovitis in my forearm.

**Comment by axs221** [comment]
I think I have a fix for this one. I will get a PR up soon. 

I can reproduce it with the mouse options disabled. Amethyst thinks that these modal type of windows are not the same as their parent window, even though they are from the same process ID. It seems to fix it if it is instead compared by process ID. I tested so far with just Chrome with Pushbullet extension, along with XCode and iTerm2 windows. Works well so far, but we will have to look out for edge cases of processes with weird PID situations.

**Comment by axs221** [comment]
Actually I am going to roll the change I mentioned into PR #654. What I was experiencing may or may not be related, and I don't know if the changes will fix the issues mentioned or not. The "is focused" logic in the Amethyst layouts was returning "false" for dialogs, and using PIDs to determine if focused seems to work better. But what I was experiencing was the window and dialog being shifted around from a new feature I added in that PR, in combination with the "is focused" problematic logic.

In other words, my fix may help your issues once #654 is merged, but I haven't experienced the same issue myself in the generally available version to verify.

**Comment by ianyh** [comment]
Solution ideas:
- Prevent applications from grabbing focus from floating windows (possibly only within the main application)
- Make floating windows actually float in the macOS sense #324

**Comment by cwisecarver** [comment]
This was happening to me with all dropdown in Jetbrains applications (PyCharm, IntelliJ, etc). Disabling "Float small windows" fixed it for me.
